### PR TITLE
Move some configuration out of sqlite, remove disk access requirement, fix package installation bugs

### LINKED ISSUE #77: steemd nodes configuration should not be stored in sqlite database
https://github.com/steemit/steem-python/blob/33dc65cc6fc85c51f57e182abe5fadff573cfa6e/steem/steemd.py#L21

### LINKED ISSUE #78: package doesn't install gracefully on macos
```    scrypt-1.2.0/libcperciva/crypto/crypto_aes.c:6:10: fatal error: 'openssl/aes.h' file not found
    #include <openssl/aes.h>
             ^~~~~~~~~~~~~~~
    1 error generated.
    error: command 'clang' failed with exit status 1
```
This is *technically* an scrypt issue, but we depend on it so we should fix it. As a workaround, I use this code in my Makefile:
```
if [[ $(shell uname) == 'Darwin' ]]; then \
    brew install openssl; \
    env LDFLAGS="-L$(brew --prefix openssl)/lib" CFLAGS="-I$(brew --prefix openssl)/include" 
fi
``` 

The installation should, at the very least, check the default hombrew openssl lib location, with an instructional error message in the event homebrew and/or openssl not installed

### LINKED ISSUE #79: package cannot be installed in docker container
https://github.com/steemit/steem-python/blob/33dc65cc6fc85c51f57e182abe5fadff573cfa6e/Pipfile#L6

Dockerfile base image doesn't support python 3.6, so build fails:

`AssertionError: steem requires Python 3.6 or newer`

### LINKED ISSUE #97: installation error in .egg
trying to install steem-python, but found this error:

Writing C:\Users\hchen\AppData\Local\Temp\easy_install-vxhgyii2\recommonmark-0.4.0\setup.cfg
Running recommonmark-0.4.0\setup.py -q bdist_egg --dist-dir C:\Users\hchen\AppData\Local\Temp\easy_install-vxhgyii2\recommonmark-0.4.0\egg-dist-tmp-z3mpobf6
File "build\bdist.win-amd64\egg\sphinxcontrib\builders\rst.py", line 139
except (IOError, OSError), err:
^
SyntaxError: invalid syntax

  File "c:\users\hchen\appdata\local\temp\pip-build-in_20u8n\steem\.eggs\sphinxcontrib_restbuilder-0.1-py3.6.egg\sphinxcontrib\builders\rst.py", line 139
    except (IOError, OSError), err:
                             ^
SyntaxError: invalid syntax

### PR DESCRIPTION
**Pull request by john-g-g**
Closes #97, closes #79, closes #78, closes #77

This commit addresses several installation bugs, in part by refactoring how the library defines its dependencies. It now uses the standard `setup.py` to define requirements, rendering many files in the package unnecessary. The refactoring also removes steemd url configuration data from sqlite, which improves installation errors that arise when install environment runtime disk write access is not desired.

### CODE REVIEW

**Comment by goldibex** [comment]
Thanks for this @john-g-g. Please do the following before review proceeds:
- The title of the PR should describe what the PR actually does.
- The first line of the description should be what is now the title. (please use the keyword `closes`, not `fixes`; this is just semantics but it's important that it all be the same)
- The rest of the description should, as briefly as possible, explain the changes introduced by the PR in plain English. In this case, where you're closing 4 issues on one PR, it should also explain why there is 1 PR for 4 issues rather than the conventional 1-to-1 relationship.

**Comment by cyon1c** [comment]
I'm going to close this @john-g-g because it's relatively outdated. The only thing I'm not clear on is the disk access requirement. If so, let's open an issue for it.
